### DISCUSSION TITLE
FastAPI converts float 1.0 to integer 1 even when using StrictFloat in a request.

### DISCUSSION THREAD

**Discussion by echatzikyriakidis** [fastapi/fastapi#7205] [Questions] [answered]
FastAPI converts float 1.0 to integer 1 even when using StrictFloat in a request.

**Comment by Kludex** [comment, marked as answer]
I didn't get what you said, so I've created a minimal example:

```python
from fastapi import FastAPI
from pydantic import StrictFloat, BaseModel

app = FastAPI()

class TestModel(BaseModel):
    test: StrictFloat


@app.post("/")
def home(what: TestModel):
    return str(type(what.test))
```
With this small app, I've inserted this body into the swagger UI:
```json
{
  "test": 1.0
}
```
And a curl command is generated (which looks like is your issue):
```bash
curl -X POST "http://localhost:8000/" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{\"test\":1}"
```
This leads to a 422:
```json
{
  "detail": [
    {
      "loc": [
        "body",
        "test"
      ],
      "msg": "value is not a valid float",
      "type": "type_error.float"
    }
  ]
}
```
But, running the right curl command on the terminal: 
```bash
curl -X POST "http://localhost:8000/" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{\"test\":1.0}"
```
I'm able to get the right response!

So basically, I think this is the problem you're facing. And I'm not sure if it's a bug. Does anyone have more knowledge about the subject?

**Comment by echatzikyriakidis** [comment]
Exactly,

I didn't noticed that Swagger UI transforms 1.0 to 1. I have tested it with CURL with correct value and it worked. Thank you!!!

For sure, there is no bug in FastAPI. Maybe in Swagger UI.

Stay safe.

**Comment by tiangolo** [comment]
Thanks for the help @Kludex ! 🍰 

And thanks for coming back to close the issue @echatzikyriakidis 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
